packages/core/contracts/proxy-scripts/lsp-broker/LspUniswapV2Broker.sol

```diff
@@ -15,27 +18,197 @@ import "../../financial-templates/long-short-pair/LongShortPair.sol";
+                FixedPoint.Signed(int256(reserveA)),
+                FixedPoint.Signed(int256(reserveB))
+            );
+            address[] memory path = new address[](2);
+
+            // 2.a) If the trade is token a to b then we are selling long tokens for short tokens. In this case, we
+            // know the exact output number of tokens we want (short tokens) and use the router's
+            // swapTokensForExactTokens with the exact output number of tokens specified.
+            if (aToB && tradeSize > 0) {
+                path[0] = address(longToken);
+                path[1] = address(shortToken);
+                TransferHelper.safeApprove(address(longToken), address(router), tradeSize);
+                router.swapTokensForExactTokens(tradeSize, type(uint256).max, path, address(this), deadline);
+            }
+            if (!aToB && tradeSize > 0) {
+                // Else, if the trade is b to a then we are selling short tokens for long tokens. In this case, we
+                // know the exact input number of tokens that we need to sell (short tokens) and can use the router's
+                // swapExactTokensForTokens with the exact input spesificed.
+                path[0] = address(shortToken);
+                path[1] = address(longToken);
+                TransferHelper.safeApprove(address(shortToken), address(router), tradeSize);
+                router.swapExactTokensForTokens(tradeSize, 0, path, address(this), deadline);
+            }
+        }
+
+        // 3) Add lequidity to the pool via the router. Approve both long and short tokens with the full account balance
+        // in this contract (i.e the amount of tokens minted +- that traded in step 2.).
+        TransferHelper.safeApprove(address(longToken), address(router), longToken.balanceOf(address(this)));
+        TransferHelper.safeApprove(address(shortToken), address(router), shortToken.balanceOf(address(this)));
+        router.addLiquidity(
+            address(longToken),
+            address(shortToken),
+            longToken.balanceOf(address(this)),
+            shortToken.balanceOf(address(this)),
+            0,
+            0,
+            address(this),
+            deadline
+        );
+        {
+            // 4) Finally, if the caller is an EOA, send back the LP tokens from minting + any dust that was left over.
+            // The dust is in the long and short tokens and will be left over as a result of small rounding errors in
+            // the calculation on how many long/short should be bought/sold to meet the pool ratio before minting.
+            if (callingAsEOA) {
+                // Send the LP tokens back to the minter.
+                IERC20 LPToken = IERC20(address(pairFor(router.factory(), address(longToken), address(shortToken))));
+                TransferHelper.safeTransfer(address(LPToken), msg.sender, LPToken.balanceOf(address(this)));
+
+                // Send any dust left over back to the minter.
```

Comment:
Can we bound or describe in more detail how `dust` actually is created? I.e. how does the algebra in `computeSwapToMintAtPoolRatio` result in any dust and why can't we get the dust to 0? 

No need to numerically bound the dust, but would be useful to know what the general source of error is (i.e. fees, unquantifiable slippage, etc.)

Reply:
I added some more comments to this effect.